src/libYARP_dev/include/yarp/dev/DataSource.h

```diff
@@ -60,14 +60,19 @@ class yarp::dev::DataWriter : public yarp::os::Runnable {
                bool canDrop=true,
                bool addStamp=false,
                IPreciselyTimed *pt=NULL) :
-        port(port), dater(dater), pPrecTime(pt), canDrop(canDrop), addStamp(addStamp)
+        port(port),
+        dater(dater), 
+        pPrecTime(pt), 
+        canDrop(canDrop), 
+        addStamp(addStamp), 
+        counter(0),
+        timePrevious(0.0),
+        cumulativeT(0.0),
+        minT(1e10),
+        maxT(0.0),
+        lastSpoke(yarp::os::Time::now())
```

Comment:
Are you sure that `yarp::os::Time::now()` and not `yarp::os::SystemClock::systemNow()` should be used here?

Reply:
Not sure about this, but copied from https://github.com/robotology/yarp/blob/7037d0b0bff7959c1c2d918dbbf238c2cf0a5dac/src/libYARP_dev/include/yarp/dev/DataSource.h#L76

Reply:
Ok!